packages/mongodb-rag-ingest/src/pages/updatePages.test.ts

```diff
@@ -66,3 +67,67 @@ describe("persistPages", () => {
     });
   });
 });
+
+describe("updatePages", () => {
+  it("fetches pages from data sources concurrently", async () => {
+    const pageStore = makeMockPageStore();
+
+    const source1: DataSource = {
+      name: "source1",
+      fetchPages: jest.fn().mockImplementation(async () => {
+        await new Promise((resolve) => setTimeout(resolve, 50)); // Simulate async delay
+        return [examplePage];
+      }),
+    } as unknown as DataSource;
+
+    const source2: DataSource = {
+      name: "source2",
+      fetchPages: jest.fn().mockImplementation(async () => {
+        await new Promise((resolve) => setTimeout(resolve, 50)); // Simulate async delay
+        return [examplePage];
+      }),
+    } as unknown as DataSource;
+    
+    const sources = [source1, source2];
+    
+    const startTimes: number[] = [];
+    const endTimes: number[] = [];
+
+    const originalDateNow = Date.now;
+    let currentFakeTime = originalDateNow();
+
+    jest.spyOn(global.Date, 'now').mockImplementation(() => currentFakeTime);
```

Comment:
similar to above on the embedded content stuff, not sure if this all is necessary. think can just do the awaited promises